Exception when compiling Java sources with SBT 1.7.0

## steps

- clone https://github.com/asflierl/java-compiler-fail.git
- run `sbt "clean;compile"`

## problem

```
> clean;compile
[info] compiling 150 Java sources to /home/asflierl/proj/java-compiler-fail/target/scala-2.13/classes ...
[error] ## Exception when compiling 150 sources to /home/asflierl/proj/java-compiler-fail/target/scala-2.13/classes
[error] java.lang.ArrayIndexOutOfBoundsException: Index 14939 out of bounds for length 61
[error] sbt.internal.inc.classfile.Parser$$anon$1.toUTF8(Parser.scala:91)
[error] sbt.internal.inc.classfile.Parser$$anon$1.parseAnnotation$1(Parser.scala:158)
[error] sbt.internal.inc.classfile.Parser$$anon$1.$anonfun$annotationsReferences$7(Parser.scala:165)
[error] scala.collection.immutable.Range.foreach$mVc$sp(Range.scala:158)
[error] sbt.internal.inc.classfile.Parser$$anon$1.$anonfun$annotationsReferences$4(Parser.scala:164)
[error] scala.collection.TraversableLike.$anonfun$flatMap$1(TraversableLike.scala:293)
[error] scala.collection.IndexedSeqOptimized.foreach(IndexedSeqOptimized.scala:36)
[error] scala.collection.IndexedSeqOptimized.foreach$(IndexedSeqOptimized.scala:33)
[error] scala.collection.mutable.ArrayOps$ofRef.foreach(ArrayOps.scala:198)
[error] scala.collection.TraversableLike.flatMap(TraversableLike.scala:293)
[error] scala.collection.TraversableLike.flatMap$(TraversableLike.scala:290)
[error] scala.collection.mutable.ArrayOps$ofRef.flatMap(ArrayOps.scala:198)
[error] sbt.internal.inc.classfile.Parser$$anon$1.annotationsReferences(Parser.scala:136)
[error] sbt.internal.inc.classfile.Parser$$anon$1.types(Parser.scala:119)
[error] sbt.internal.inc.classfile.JavaAnalyze$.$anonfun$apply$31(JavaAnalyze.scala:177)
[error] scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:286)
[error] scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
[error] scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
[error] scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
[error] scala.collection.TraversableLike.map(TraversableLike.scala:286)
[error] scala.collection.TraversableLike.map$(TraversableLike.scala:279)
[error] scala.collection.AbstractTraversable.map(Traversable.scala:108)
[error] sbt.internal.inc.classfile.JavaAnalyze$.$anonfun$apply$16(JavaAnalyze.scala:177)
[error] sbt.internal.inc.classfile.JavaAnalyze$.$anonfun$apply$16$adapted(JavaAnalyze.scala:108)
[error] scala.collection.TraversableLike$WithFilter.$anonfun$foreach$1(TraversableLike.scala:985)
[error] scala.collection.mutable.HashMap.$anonfun$foreach$1(HashMap.scala:149)
[error] scala.collection.mutable.HashTable.foreachEntry(HashTable.scala:237)
[error] scala.collection.mutable.HashTable.foreachEntry$(HashTable.scala:230)
[error] scala.collection.mutable.HashMap.foreachEntry(HashMap.scala:44)
[error] scala.collection.mutable.HashMap.foreach(HashMap.scala:149)
[error] scala.collection.TraversableLike$WithFilter.foreach(TraversableLike.scala:984)
[error] sbt.internal.inc.classfile.JavaAnalyze$.apply(JavaAnalyze.scala:108)
[error] sbt.internal.inc.javac.AnalyzingJavaCompiler.$anonfun$compile$21(AnalyzingJavaCompiler.scala:215)
[error] sbt.internal.inc.javac.AnalyzingJavaCompiler.$anonfun$compile$21$adapted(AnalyzingJavaCompiler.scala:207)
[error] scala.collection.TraversableLike$WithFilter.$anonfun$foreach$1(TraversableLike.scala:985)
[error] scala.collection.immutable.List.foreach(List.scala:431)
[error] scala.collection.TraversableLike$WithFilter.foreach(TraversableLike.scala:984)
[error] sbt.internal.inc.javac.AnalyzingJavaCompiler.$anonfun$compile$19(AnalyzingJavaCompiler.scala:207)
[error] scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[error] sbt.internal.inc.javac.AnalyzingJavaCompiler.timed(AnalyzingJavaCompiler.scala:262)
[error] sbt.internal.inc.javac.AnalyzingJavaCompiler.compile(AnalyzingJavaCompiler.scala:207)
[error] sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compileJava$2(MixedAnalyzingCompiler.scala:103)
[error] sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compileJava$2$adapted(MixedAnalyzingCompiler.scala:91)
[error] sbt.internal.inc.JarUtils$.withPreviousJar(JarUtils.scala:239)
[error] sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compileJava$1(MixedAnalyzingCompiler.scala:91)
[error] scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[error] sbt.internal.inc.MixedAnalyzingCompiler.timed(MixedAnalyzingCompiler.scala:248)
[error] sbt.internal.inc.MixedAnalyzingCompiler.compileJava(MixedAnalyzingCompiler.scala:61)
[error] sbt.internal.inc.MixedAnalyzingCompiler.compileJava0$1(MixedAnalyzingCompiler.scala:198)
[error] sbt.internal.inc.MixedAnalyzingCompiler.compile(MixedAnalyzingCompiler.scala:211)
[error] sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1(IncrementalCompilerImpl.scala:534)
[error] sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1$adapted(IncrementalCompilerImpl.scala:534)
[error] sbt.internal.inc.Incremental$.$anonfun$apply$5(Incremental.scala:179)
[error] sbt.internal.inc.Incremental$.$anonfun$apply$5$adapted(Incremental.scala:177)
[error] sbt.internal.inc.Incremental$$anon$2.run(Incremental.scala:463)
[error] sbt.internal.inc.IncrementalCommon$CycleState.next(IncrementalCommon.scala:116)
[error] sbt.internal.inc.IncrementalCommon$$anon$1.next(IncrementalCommon.scala:56)
[error] sbt.internal.inc.IncrementalCommon$$anon$1.next(IncrementalCommon.scala:52)
[error] sbt.internal.inc.IncrementalCommon.cycle(IncrementalCommon.scala:263)
[error] sbt.internal.inc.Incremental$.$anonfun$incrementalCompile$8(Incremental.scala:418)
[error] sbt.internal.inc.Incremental$.withClassfileManager(Incremental.scala:506)
[error] sbt.internal.inc.Incremental$.incrementalCompile(Incremental.scala:405)
[error] sbt.internal.inc.Incremental$.apply(Incremental.scala:171)
[error] sbt.internal.inc.IncrementalCompilerImpl.compileInternal(IncrementalCompilerImpl.scala:534)
[error] sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileIncrementally$1(IncrementalCompilerImpl.scala:488)
[error] sbt.internal.inc.IncrementalCompilerImpl.handleCompilationError(IncrementalCompilerImpl.scala:332)
[error] sbt.internal.inc.IncrementalCompilerImpl.compileIncrementally(IncrementalCompilerImpl.scala:425)
[error] sbt.internal.inc.IncrementalCompilerImpl.compile(IncrementalCompilerImpl.scala:137)
[error] sbt.Defaults$.compileIncrementalTaskImpl(Defaults.scala:2364)
[error] sbt.Defaults$.$anonfun$compileIncrementalTask$2(Defaults.scala:2314)
[error] sbt.internal.server.BspCompileTask$.$anonfun$compute$1(BspCompileTask.scala:30)
[error] sbt.internal.io.Retry$.apply(Retry.scala:46)
[error] sbt.internal.io.Retry$.apply(Retry.scala:28)
[error] sbt.internal.io.Retry$.apply(Retry.scala:23)
[error] sbt.internal.server.BspCompileTask$.compute(BspCompileTask.scala:30)
[error] sbt.Defaults$.$anonfun$compileIncrementalTask$1(Defaults.scala:2312)
[error] scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error] sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:62)
[error] sbt.std.Transform$$anon$4.work(Transform.scala:68)
[error] sbt.Execute.$anonfun$submit$2(Execute.scala:282)
[error] sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:23)
[error] sbt.Execute.work(Execute.scala:291)
[error] sbt.Execute.$anonfun$submit$1(Execute.scala:282)
[error] sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
[error] sbt.CompletionService$$anon$2.call(CompletionService.scala:64)
[error] java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error] java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
[error] java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error] java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[error] java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[error] java.base/java.lang.Thread.run(Thread.java:833)
[error]
[error] stack trace is suppressed; run last Compile / compileIncremental for the full output
[error] (Compile / compileIncremental) java.lang.ArrayIndexOutOfBoundsException: Index 14939 out of bounds for length 61
```

## expectation

Java sources are compiled without compiler exception.

## notes

I tried this with SBT 1.6.2 and SBT 1.7.0-RC2 and both worked. Only occurs with SBT 1.7.0.
I am using JDK 17.

somthing like that with this exception:

```
[error] ## Exception when compiling 282 sources to /home/me/sbt-proj/target/scala-2.13/classes
[error] java.lang.AssertionError: assumption failed: Constant pool entry is not a UTF8 type: 254
[error] scala.Predef$.assume(Predef.scala:256)
[error] sbt.internal.inc.classfile.Parser$$anon$1.toUTF8(Parser.scala:92)
[error] sbt.internal.inc.classfile.Parser$$anon$1.parseAnnotation$1(Parser.scala:158)
[error] sbt.internal.inc.classfile.Parser$$anon$1.$anonfun$annotationsReferences$7(Parser.scala:165)
[error] scala.collection.immutable.Range.foreach$mVc$sp(Range.scala:158)
[error] sbt.internal.inc.classfile.Parser$$anon$1.$anonfun$annotationsReferences$4(Parser.scala:164)
[error] scala.collection.TraversableLike.$anonfun$flatMap$1(TraversableLike.scala:293)
[error] scala.collection.IndexedSeqOptimized.foreach(IndexedSeqOptimized.scala:36)
[error] scala.collection.IndexedSeqOptimized.foreach$(IndexedSeqOptimized.scala:33)
[error] scala.collection.mutable.ArrayOps$ofRef.foreach(ArrayOps.scala:198)
[error] scala.collection.TraversableLike.flatMap(TraversableLike.scala:293)
[error] scala.collection.TraversableLike.flatMap$(TraversableLike.scala:290)
[error] scala.collection.mutable.ArrayOps$ofRef.flatMap(ArrayOps.scala:198)
[error] sbt.internal.inc.classfile.Parser$$anon$1.annotationsReferences(Parser.scala:136)
[error] sbt.internal.inc.classfile.Parser$$anon$1.types(Parser.scala:119)
[error] sbt.internal.inc.classfile.JavaAnalyze$.$anonfun$apply$31(JavaAnalyze.scala:177)
[error] scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:286)
[error] scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
[error] scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
[error] scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
[error] scala.collection.TraversableLike.map(TraversableLike.scala:286)
[error] scala.collection.TraversableLike.map$(TraversableLike.scala:279)
[error] scala.collection.AbstractTraversable.map(Traversable.scala:108)
[error] sbt.internal.inc.classfile.JavaAnalyze$.$anonfun$apply$16(JavaAnalyze.scala:177)
[error] sbt.internal.inc.classfile.JavaAnalyze$.$anonfun$apply$16$adapted(JavaAnalyze.scala:108)
[error] scala.collection.TraversableLike$WithFilter.$anonfun$foreach$1(TraversableLike.scala:985)
[error] scala.collection.mutable.HashMap.$anonfun$foreach$1(HashMap.scala:149)
[error] scala.collection.mutable.HashTable.foreachEntry(HashTable.scala:237)
[error] scala.collection.mutable.HashTable.foreachEntry$(HashTable.scala:230)
[error] scala.collection.mutable.HashMap.foreachEntry(HashMap.scala:44)
[error] scala.collection.mutable.HashMap.foreach(HashMap.scala:149)
[error] scala.collection.TraversableLike$WithFilter.foreach(TraversableLike.scala:984)
[error] sbt.internal.inc.classfile.JavaAnalyze$.apply(JavaAnalyze.scala:108)
[error] sbt.internal.inc.javac.AnalyzingJavaCompiler.$anonfun$compile$21(AnalyzingJavaCompiler.scala:215)
[error] sbt.internal.inc.javac.AnalyzingJavaCompiler.$anonfun$compile$21$adapted(AnalyzingJavaCompiler.scala:207)
[error] scala.collection.TraversableLike$WithFilter.$anonfun$foreach$1(TraversableLike.scala:985)
[error] scala.collection.immutable.List.foreach(List.scala:431)
[error] scala.collection.TraversableLike$WithFilter.foreach(TraversableLike.scala:984)
[error] sbt.internal.inc.javac.AnalyzingJavaCompiler.$anonfun$compile$19(AnalyzingJavaCompiler.scala:207)
[error] scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[error] sbt.internal.inc.javac.AnalyzingJavaCompiler.timed(AnalyzingJavaCompiler.scala:262)
[error] sbt.internal.inc.javac.AnalyzingJavaCompiler.compile(AnalyzingJavaCompiler.scala:207)
[error] sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compileJava$2(MixedAnalyzingCompiler.scala:103)
[error] sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compileJava$2$adapted(MixedAnalyzingCompiler.scala:91)
[error] sbt.internal.inc.JarUtils$.withPreviousJar(JarUtils.scala:239)
[error] sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compileJava$1(MixedAnalyzingCompiler.scala:91)
[error] scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[error] sbt.internal.inc.MixedAnalyzingCompiler.timed(MixedAnalyzingCompiler.scala:248)
[error] sbt.internal.inc.MixedAnalyzingCompiler.compileJava(MixedAnalyzingCompiler.scala:61)
[error] sbt.internal.inc.MixedAnalyzingCompiler.compileJava0$1(MixedAnalyzingCompiler.scala:198)
[error] sbt.internal.inc.MixedAnalyzingCompiler.compile(MixedAnalyzingCompiler.scala:211)
[error] sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1(IncrementalCompilerImpl.scala:534)
[error] sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1$adapted(IncrementalCompilerImpl.scala:534)
[error] sbt.internal.inc.Incremental$.$anonfun$apply$5(Incremental.scala:179)
[error] sbt.internal.inc.Incremental$.$anonfun$apply$5$adapted(Incremental.scala:177)
[error] sbt.internal.inc.Incremental$$anon$2.run(Incremental.scala:463)
[error] sbt.internal.inc.IncrementalCommon$CycleState.next(IncrementalCommon.scala:116)
[error] sbt.internal.inc.IncrementalCommon$$anon$1.next(IncrementalCommon.scala:56)
[error] sbt.internal.inc.IncrementalCommon$$anon$1.next(IncrementalCommon.scala:52)
[error] sbt.internal.inc.IncrementalCommon.cycle(IncrementalCommon.scala:263)
[error] sbt.internal.inc.Incremental$.$anonfun$incrementalCompile$8(Incremental.scala:418)
[error] sbt.internal.inc.Incremental$.withClassfileManager(Incremental.scala:506)
[error] sbt.internal.inc.Incremental$.incrementalCompile(Incremental.scala:405)
[error] sbt.internal.inc.Incremental$.apply(Incremental.scala:171)
[error] sbt.internal.inc.IncrementalCompilerImpl.compileInternal(IncrementalCompilerImpl.scala:534)
[error] sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileIncrementally$1(IncrementalCompilerImpl.scala:488)
[error] sbt.internal.inc.IncrementalCompilerImpl.handleCompilationError(IncrementalCompilerImpl.scala:332)
[error] sbt.internal.inc.IncrementalCompilerImpl.compileIncrementally(IncrementalCompilerImpl.scala:425)
[error] sbt.internal.inc.IncrementalCompilerImpl.compile(IncrementalCompilerImpl.scala:137)
[error] sbt.Defaults$.compileIncrementalTaskImpl(Defaults.scala:2364)
[error] sbt.Defaults$.$anonfun$compileIncrementalTask$2(Defaults.scala:2314)
[error] sbt.internal.server.BspCompileTask$.$anonfun$compute$1(BspCompileTask.scala:30)
[error] sbt.internal.io.Retry$.apply(Retry.scala:46)
[error] sbt.internal.io.Retry$.apply(Retry.scala:28)
[error] sbt.internal.io.Retry$.apply(Retry.scala:23)
[error] sbt.internal.server.BspCompileTask$.compute(BspCompileTask.scala:30)
[error] sbt.Defaults$.$anonfun$compileIncrementalTask$1(Defaults.scala:2312)
[error] scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error] sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:62)
[error] sbt.std.Transform$$anon$4.work(Transform.scala:68)
[error] sbt.Execute.$anonfun$submit$2(Execute.scala:282)
[error] sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:23)
[error] sbt.Execute.work(Execute.scala:291)
[error] sbt.Execute.$anonfun$submit$1(Execute.scala:282)
[error] sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
[error] sbt.CompletionService$$anon$2.call(CompletionService.scala:64)
[error] java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error] java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
[error] java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error] java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[error] java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[error] java.base/java.lang.Thread.run(Thread.java:833)
[error]            
[error] java.lang.AssertionError: assumption failed: Constant pool entry is not a UTF8 type: 254
[error] 	at scala.Predef$.assume(Predef.scala:256)
[error] 	at sbt.internal.inc.classfile.Parser$$anon$1.toUTF8(Parser.scala:92)
[error] 	at sbt.internal.inc.classfile.Parser$$anon$1.parseAnnotation$1(Parser.scala:158)
[error] 	at sbt.internal.inc.classfile.Parser$$anon$1.$anonfun$annotationsReferences$7(Parser.scala:165)
[error] 	at scala.collection.immutable.Range.foreach$mVc$sp(Range.scala:158)
[error] 	at sbt.internal.inc.classfile.Parser$$anon$1.$anonfun$annotationsReferences$4(Parser.scala:164)
[error] 	at scala.collection.TraversableLike.$anonfun$flatMap$1(TraversableLike.scala:293)
[error] 	at scala.collection.IndexedSeqOptimized.foreach(IndexedSeqOptimized.scala:36)
[error] 	at scala.collection.IndexedSeqOptimized.foreach$(IndexedSeqOptimized.scala:33)
[error] 	at scala.collection.mutable.ArrayOps$ofRef.foreach(ArrayOps.scala:198)
[error] 	at scala.collection.TraversableLike.flatMap(TraversableLike.scala:293)
[error] 	at scala.collection.TraversableLike.flatMap$(TraversableLike.scala:290)
[error] 	at scala.collection.mutable.ArrayOps$ofRef.flatMap(ArrayOps.scala:198)
[error] 	at sbt.internal.inc.classfile.Parser$$anon$1.annotationsReferences(Parser.scala:136)
[error] 	at sbt.internal.inc.classfile.Parser$$anon$1.types(Parser.scala:119)
[error] 	at sbt.internal.inc.classfile.JavaAnalyze$.$anonfun$apply$31(JavaAnalyze.scala:177)
[error] 	at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:286)
[error] 	at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
[error] 	at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
[error] 	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
[error] 	at scala.collection.TraversableLike.map(TraversableLike.scala:286)
[error] 	at scala.collection.TraversableLike.map$(TraversableLike.scala:279)
[error] 	at scala.collection.AbstractTraversable.map(Traversable.scala:108)
[error] 	at sbt.internal.inc.classfile.JavaAnalyze$.$anonfun$apply$16(JavaAnalyze.scala:177)
[error] 	at sbt.internal.inc.classfile.JavaAnalyze$.$anonfun$apply$16$adapted(JavaAnalyze.scala:108)
[error] 	at scala.collection.TraversableLike$WithFilter.$anonfun$foreach$1(TraversableLike.scala:985)
[error] 	at scala.collection.mutable.HashMap.$anonfun$foreach$1(HashMap.scala:149)
[error] 	at scala.collection.mutable.HashTable.foreachEntry(HashTable.scala:237)
[error] 	at scala.collection.mutable.HashTable.foreachEntry$(HashTable.scala:230)
[error] 	at scala.collection.mutable.HashMap.foreachEntry(HashMap.scala:44)
[error] 	at scala.collection.mutable.HashMap.foreach(HashMap.scala:149)
[error] 	at scala.collection.TraversableLike$WithFilter.foreach(TraversableLike.scala:984)
[error] 	at sbt.internal.inc.classfile.JavaAnalyze$.apply(JavaAnalyze.scala:108)
[error] 	at sbt.internal.inc.javac.AnalyzingJavaCompiler.$anonfun$compile$21(AnalyzingJavaCompiler.scala:215)
[error] 	at sbt.internal.inc.javac.AnalyzingJavaCompiler.$anonfun$compile$21$adapted(AnalyzingJavaCompiler.scala:207)
[error] 	at scala.collection.TraversableLike$WithFilter.$anonfun$foreach$1(TraversableLike.scala:985)
[error] 	at scala.collection.immutable.List.foreach(List.scala:431)
[error] 	at scala.collection.TraversableLike$WithFilter.foreach(TraversableLike.scala:984)
[error] 	at sbt.internal.inc.javac.AnalyzingJavaCompiler.$anonfun$compile$19(AnalyzingJavaCompiler.scala:207)
[error] 	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[error] 	at sbt.internal.inc.javac.AnalyzingJavaCompiler.timed(AnalyzingJavaCompiler.scala:262)
[error] 	at sbt.internal.inc.javac.AnalyzingJavaCompiler.compile(AnalyzingJavaCompiler.scala:207)
[error] 	at sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compileJava$2(MixedAnalyzingCompiler.scala:103)
[error] 	at sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compileJava$2$adapted(MixedAnalyzingCompiler.scala:91)
[error] 	at sbt.internal.inc.JarUtils$.withPreviousJar(JarUtils.scala:239)
[error] 	at sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compileJava$1(MixedAnalyzingCompiler.scala:91)
[error] 	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[error] 	at sbt.internal.inc.MixedAnalyzingCompiler.timed(MixedAnalyzingCompiler.scala:248)
[error] 	at sbt.internal.inc.MixedAnalyzingCompiler.compileJava(MixedAnalyzingCompiler.scala:61)
[error] 	at sbt.internal.inc.MixedAnalyzingCompiler.compileJava0$1(MixedAnalyzingCompiler.scala:198)
[error] 	at sbt.internal.inc.MixedAnalyzingCompiler.compile(MixedAnalyzingCompiler.scala:211)
[error] 	at sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1(IncrementalCompilerImpl.scala:534)
[error] 	at sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1$adapted(IncrementalCompilerImpl.scala:534)
[error] 	at sbt.internal.inc.Incremental$.$anonfun$apply$5(Incremental.scala:179)
[error] 	at sbt.internal.inc.Incremental$.$anonfun$apply$5$adapted(Incremental.scala:177)
[error] 	at sbt.internal.inc.Incremental$$anon$2.run(Incremental.scala:463)
[error] 	at sbt.internal.inc.IncrementalCommon$CycleState.next(IncrementalCommon.scala:116)
[error] 	at sbt.internal.inc.IncrementalCommon$$anon$1.next(IncrementalCommon.scala:56)
[error] 	at sbt.internal.inc.IncrementalCommon$$anon$1.next(IncrementalCommon.scala:52)
[error] 	at sbt.internal.inc.IncrementalCommon.cycle(IncrementalCommon.scala:263)
[error] 	at sbt.internal.inc.Incremental$.$anonfun$incrementalCompile$8(Incremental.scala:418)
[error] 	at sbt.internal.inc.Incremental$.withClassfileManager(Incremental.scala:506)
[error] 	at sbt.internal.inc.Incremental$.incrementalCompile(Incremental.scala:405)
[error] 	at sbt.internal.inc.Incremental$.apply(Incremental.scala:171)
[error] 	at sbt.internal.inc.IncrementalCompilerImpl.compileInternal(IncrementalCompilerImpl.scala:534)
[error] 	at sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileIncrementally$1(IncrementalCompilerImpl.scala:488)
[error] 	at sbt.internal.inc.IncrementalCompilerImpl.handleCompilationError(IncrementalCompilerImpl.scala:332)
[error] 	at sbt.internal.inc.IncrementalCompilerImpl.compileIncrementally(IncrementalCompilerImpl.scala:425)
[error] 	at sbt.internal.inc.IncrementalCompilerImpl.compile(IncrementalCompilerImpl.scala:137)
[error] 	at sbt.Defaults$.compileIncrementalTaskImpl(Defaults.scala:2364)
[error] 	at sbt.Defaults$.$anonfun$compileIncrementalTask$2(Defaults.scala:2314)
[error] 	at sbt.internal.server.BspCompileTask$.$anonfun$compute$1(BspCompileTask.scala:30)
[error] 	at sbt.internal.io.Retry$.apply(Retry.scala:46)
[error] 	at sbt.internal.io.Retry$.apply(Retry.scala:28)
[error] 	at sbt.internal.io.Retry$.apply(Retry.scala:23)
[error] 	at sbt.internal.server.BspCompileTask$.compute(BspCompileTask.scala:30)
[error] 	at sbt.Defaults$.$anonfun$compileIncrementalTask$1(Defaults.scala:2312)
[error] 	at scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error] 	at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:62)
[error] 	at sbt.std.Transform$$anon$4.work(Transform.scala:68)
[error] 	at sbt.Execute.$anonfun$submit$2(Execute.scala:282)
[error] 	at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:23)
[error] 	at sbt.Execute.work(Execute.scala:291)
[error] 	at sbt.Execute.$anonfun$submit$1(Execute.scala:282)
[error] 	at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
[error] 	at sbt.CompletionService$$anon$2.call(CompletionService.scala:64)
[error] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error] 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
[error] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error] 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[error] 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[error] 	at java.base/java.lang.Thread.run(Thread.java:833)
[error] (Compile / compileIncremental) java.lang.AssertionError: assumption failed: Constant pool entry is not a UTF8 type: 254

```

work properly with 1.6.2 and 1.7.0-RC2

Can't compile Play with sbt 1.7 as well: https://github.com/playframework/playframework/pull/11349

https://github.com/playframework/playframework/runs/7281244256?check_suite_focus=true#step:9:5395:
```
java.lang.AssertionError: assumption failed: Constant pool entry is not a UTF8 type: 13
````

--- a/internal/zinc-classfile/src/main/scala/sbt/internal/inc/classfile/Parser.scala
+++ b/internal/zinc-classfile/src/main/scala/sbt/internal/inc/classfile/Parser.scala
@@ -164,8 +164,12 @@ private[sbt] object Parser {
                 case '[' =>
                   for (_ <- 0 until in.readUnsignedShort())
                     parseElementValue()
+                case 'B' | 'C' | 'D' | 'F' | 'I' | 'J' | 'S' | 'Z' | 's' =>
+                  val _ = in.readUnsignedShort()
                 case _ =>
-                // ignore. robustness is paramount
+                  // if we see something unexpected, we're likely already doomed and trying to
+                  // continue parsing will just make troubleshooting harder. so let's bail
+                  sys.error(s"unexpected tag in annotation: '$c'")
               }
             }
             def parseAnnotation(): Unit = { // JVMS 4.7.16

--- a/internal/zinc-classfile/src/main/scala/sbt/internal/inc/classfile/Parser.scala
+++ b/internal/zinc-classfile/src/main/scala/sbt/internal/inc/classfile/Parser.scala
@@ -115,8 +115,22 @@ private[sbt] object Parser {
         AttributeInfo(name, value)
       }
 
-      def types =
-        (classConstantReferences ++ fieldTypes ++ methodTypes ++ annotationsReferences).toSet
+      def types: Set[String] = {
+        // support for parsing annotations in Java classfiles is new in sbt 1.7.0.  as of July 2022
+        // we don't completely trust it yet (one bug already forced us to build sbt 1.7.1) so if
+        // that part of the parser blows up, we only warn rather than failing compilation. this runs
+        // the risk of the warning not being noticed and the bug not being reported, but so be it.
+        // (for the time being, at least!)
+        def annotationsReferencesCarefully =
+          try annotationsReferences
+          catch {
+            case re: RuntimeException =>
+              log.warn(s"couldn't parse annotations in $readableName ($re)")
+              List()
+          }
+        // the other aspects of classfile.Parser are long battle-tested
+        (classConstantReferences ++ fieldTypes ++ methodTypes ++ annotationsReferencesCarefully).toSet
+      }
 
       private def getTypes(fieldsOrMethods: Array[FieldOrMethodInfo]) =
         fieldsOrMethods.flatMap { fieldOrMethod =>

--- a/internal/zinc-classfile/src/test/scala/sbt/internal/inc/classfile/AnalyzeSpecification.scala
+++ b/internal/zinc-classfile/src/test/scala/sbt/internal/inc/classfile/AnalyzeSpecification.scala
@@ -184,4 +184,34 @@ class AnalyzeSpecification extends UnitSpec {
     assert(deps.memberRef("Foo").contains("Test"))
   }
 
+  // issue: sbt/sbt#6969
+  "Analyze" should "handle multiple annotations on field" in {
+    val srcA1 =
+      """|import java.lang.annotation.Retention;
+         |import java.lang.annotation.RetentionPolicy;
+         |@Retention(RetentionPolicy.RUNTIME)
+         |public @interface A1 {
+         |  String s();
+         |}
+         |""".stripMargin
+    val srcA2 =
+      """|import java.lang.annotation.Retention;
+         |import java.lang.annotation.RetentionPolicy;
+         |@Retention(RetentionPolicy.RUNTIME)
+         |public @interface A2 { }
+         |""".stripMargin
+    val srcFoo =
+      """|public class Foo {
+         |  @A1(s = "id") @A2 String id;
+         |}
+         |""".stripMargin
+    val deps = JavaCompilerForUnitTesting.extractDependenciesFromSrcs(
+      "A1.java" -> srcA1,
+      "A2.java" -> srcA2,
+      "Foo.java" -> srcFoo,
+    )
+    assert(deps.memberRef("Foo").contains("A1"))
+    assert(deps.memberRef("Foo").contains("A2"))
+  }
+
 }

